File: day04/src/test/java/de/havox_design/aoc2015/day04/Day04Test.java
package de.havox_design.aoc2015.day04;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day04Test {

    @SuppressWarnings("java:S2699")
    @Test
    void testMainClass() {
        MainClass.main(new String[0]);
    }

    @ParameterizedTest
    @MethodSource("getDataForPart1")
    void testPart1(String fileName, int expectedSecretNumber) {
        Assertions.assertEquals(expectedSecretNumber, AdventCoins.solvePart1(fileName));
    }

    private static Stream<Arguments> getDataForPart1() {
        return Stream.of(
                Arguments.of("part1sample1.txt", 609043),
                Arguments.of("part1sample2.txt", 1048970)
        );
    }
}

File: day04/src/test/kotlin/de/havox_design/aoc2023/day04/Day04Test.kt
package de.havox_design.aoc2023.day04

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class Day04Test {
    @Test
    fun testMainClass() {
        MainClass.main(arrayOf())
    }
}

private fun String.shouldBe(expectation: String) = assertEquals(expectation, this)